### PR TITLE
Change to cli

### PR DESCRIPTION
**Pull request by Random-Liu**
Use cobra for cli. With this change, the command will be like:
```console
$ ./_output/cri-containerd --help
             _                         __        _                      __
  __________(_)      _________  ____  / /_____ _(_)____  ___  _________/ /
 / ___/ ___/ /______/ ___/ __ \/ __ \/ __/ __ `/ // __ \/ _ \/ ___/ __  /
/ /__/ /  / //_____/ /__/ /_/ / / / / /_/ /_/ / // / / /  __/ /  / /_/ /
\___/_/  /_/       \___/\____/_/ /_/\__/\__,_/_//_/ /_/\___/_/   \__,_/

A containerd based Kubernetes CRI implementation.

Usage:
  cri-containerd [flags]
  cri-containerd [command]

Available Commands:
  default-config Print default toml config of cri-containerd.
  version        Print cri-containerd version information.

Flags:
      --alsologtostderr                    log to standard error as well as files
      --cgroup-path string                 The cgroup that cri-containerd is part of. Cri-containerd is not placed in a cgroup if none is specified.
      --config string                      Path to the config file. (default "/etc/cri-containerd/config.toml")
      --containerd-endpoint string         Path to the containerd endpoint. (default "/run/containerd/containerd.sock")
      --containerd-root-dir string         Root directory path where containerd stores persistent data. (default "/var/lib/containerd")
      --containerd-runtime string          The runtime used by containerd. (default "io.containerd.runtime.v1.linux")
      --containerd-runtime-engine string   Runtime engine used by containerd. Defaults to containerd's default if not specified.
      --containerd-runtime-root string     The directory used by containerd for runtime state. Defaults to containerd's default if not specified.
      --containerd-snapshotter string      The snapshotter used by containerd. (default "overlayfs")
      --enable-selinux                     Enable selinux support. By default not enabled.
      --log_backtrace_at traceLocation     when logging hits line file:N, emit a stack trace (default :0)
      --log_dir string                     If non-empty, write log files in this directory
      --logtostderr                        log to standard error instead of files
      --network-bin-dir string             The directory for putting network binaries. (default "/opt/cni/bin")
      --network-conf-dir string            The directory for putting network plugin configuration files. (default "/etc/cni/net.d")
      --oom-score int                      Adjust the cri-containerd's oom score. (default -999)
      --root-dir string                    Root directory path for cri-containerd managed files (metadata checkpoint etc). (default "/var/lib/cri-containerd")
      --sandbox-image string               The image used by sandbox container. (default "gcr.io/google_containers/pause:3.0")
      --socket-path string                 Path to the socket which cri-containerd serves on. (default "/var/run/cri-containerd.sock")
      --stats-collect-period int           The period (in seconds) of snapshots stats collection. (default 10)
      --stderrthreshold severity           logs at or above this threshold go to stderr (default 2)
      --stream-addr string                 The ip address streaming server is listening on. The default host interface is used if not specified.
      --stream-port string                 The port streaming server is listening on. (default "10010")
      --systemd-cgroup                     Enables systemd cgroup support. By default not enabled.
  -v, --v Level                            log level for V logs
      --vmodule moduleSpec                 comma-separated list of pattern=N settings for file-filtered logging

Use "cri-containerd [command] --help" for more information about a command.
$ _output/cri-containerd version
1.0.0-alpha.0-55-gc056fe3-dirty
$ _output/cri-containerd default-config
socket_path = "/var/run/cri-containerd.sock"
root_dir = "/var/lib/cri-containerd"
stream_server_address = ""
stream_server_port = "10010"
cgroup_path = ""
enable_selinux = false
sandbox_image = "gcr.io/google_containers/pause:3.0"
stats_collect_period = 10
systemd_cgroup = false
oom_score = -999

[containerd]
  root_dir = "/var/lib/containerd"
  snapshotter = "overlayfs"
  endpoint = "/run/containerd/containerd.sock"
  runtime = "io.containerd.runtime.v1.linux"
  runtime_engine = ""
  runtime_root = ""

[cni]
  bin_dir = "/opt/cni/bin"
  conf_dir = "/etc/cni/net.d"
```
@mikebrow @yanxuean

### CODE REVIEW

**Comment by Random-Liu** [comment]
I'll add image load sub command based on this PR.

File: cmd/cri-containerd/cri_containerd.go [hunk 1]
```diff
@@ -17,53 +17,95 @@ limitations under the License.
...
-	glog.V(2).Infof("Run cri-containerd grpc server on socket %q", o.SocketPath)
-	s, err := server.NewCRIContainerdService(o.Config)
-	if err != nil {
-		glog.Exitf("Failed to create CRI containerd service: %v", err)
+		glog.V(2).Infof("Run cri-containerd grpc server on socket %q", o.SocketPath)
+		s, err := server.NewCRIContainerdService(o.Config)
+		if err != nil {
+			glog.Exitf("Failed to create CRI containerd service: %v", err)
+		}
+		// Use interrupt handler to make sure the server to be stopped properly.
```
Comment (mikebrow) [inline cmd/cri-containerd/cri_containerd.go]: /s/to be/is/

File: cmd/cri-containerd/options/options.go [hunk 2]
```diff
@@ -105,63 +100,52 @@ func NewCRIContainerdOptions() *CRIContainerdOptions {
 
 // AddFlags adds cri-containerd command line options to pflag.
 func (c *CRIContainerdOptions) AddFlags(fs *pflag.FlagSet) {
+	defaults := defaultConfig()
 	fs.StringVar(&c.ConfigFilePath, configFilePathArgName,
 		"/etc/cri-containerd/config.toml", "Path to the config file.")
```
Comment (mikebrow) [inline cmd/cri-containerd/options/options.go]: defaults.ConfigFilePath?

File: cmd/cri-containerd/options/options.go [hunk 3]
```diff
@@ -179,17 +163,41 @@ func (c *CRIContainerdOptions) InitFlags(fs *pflag.FlagSet) error {
 	// But the priority of the toml config value is higher than the default value,
 	// Without a way to insert the toml config value between the default value and the command line value.
 	// We parse twice one for default value, one for commandline value.
-	return fs.Parse(commandline)
+	return fs.Parse(os.Args[1:])
```
Comment (mikebrow) [inline cmd/cri-containerd/options/options.go]: whoot! one parse :-)

File: cmd/cri-containerd/cri_containerd.go (hunk 1 above)
Comment (Random-Liu) [reply cmd/cri-containerd/cri_containerd.go]: Will do.

File: cmd/cri-containerd/options/options.go (hunk 2 above)
Comment (Random-Liu) [reply cmd/cri-containerd/options/options.go]: `ConfigFilePath` is not part of config and default config. But I can make this a const. :)

File: cmd/cri-containerd/cri_containerd.go (hunk 1 above)
Comment (Random-Liu) [reply cmd/cri-containerd/cri_containerd.go]: Done.

**Comment by Random-Liu** [comment]
@mikebrow Done.

**Comment by abhi** [comment]
LGTM

File: cmd/cri-containerd/options/options.go [hunk 4]
```diff
@@ -105,63 +104,52 @@ func NewCRIContainerdOptions() *CRIContainerdOptions {
...
 func (c *CRIContainerdOptions) InitFlags(fs *pflag.FlagSet) error {
-	fs.AddGoFlagSet(flag.CommandLine)
-
-	commandline := os.Args[1:]
-	if err := fs.Parse(commandline); err != nil {
-		return err
-	}
-
 	// Load default config file if none provided
 	if _, err := toml.DecodeFile(c.ConfigFilePath, &c.Config); err != nil {
```
Comment (yanxuean) [inline cmd/cri-containerd/options/options.go]: Can we modify ConfigFilePath to specification?

File: cmd/cri-containerd/options/options.go [hunk 5]
```diff
@@ -179,17 +167,41 @@ func (c *CRIContainerdOptions) InitFlags(fs *pflag.FlagSet) error {
 	// But the priority of the toml config value is higher than the default value,
 	// Without a way to insert the toml config value between the default value and the command line value.
 	// We parse twice one for default value, one for commandline value.
```
Comment (yanxuean) [inline cmd/cri-containerd/options/options.go]: If we just need one parse, The comment is outdated.

Comment (Random-Liu) [reply cmd/cri-containerd/options/options.go]: The first parse is done by cobra, so actually there are still 2 parse.

File: cmd/cri-containerd/options/options.go (hunk 4 above)
Comment (Random-Liu) [reply cmd/cri-containerd/options/options.go]: What do you mean?

File: cmd/cri-containerd/options/options.go (hunk 5 above)
Comment (yanxuean) [reply cmd/cri-containerd/options/options.go]: I see.   cobra.Command.Execute() do it. 
Thanks

File: cmd/cri-containerd/options/options.go (hunk 4 above)
Comment (yanxuean) [reply cmd/cri-containerd/options/options.go]: You has answered it at below. Tks

**Pull request merged**